src/main/java/org/openrewrite/java/logging/slf4j/StripToStringFromArguments.java

```diff
@@ -0,0 +1,59 @@
+package org.openrewrite.java.logging.slf4j;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class StripToStringFromArguments extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Strip `toString()` from arguments";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Remove `.toString()` from logger call arguments; SLF4J will automatically call `toString()` on an argument when not a string, and do so only if the log level is enabled.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(Preconditions.or(
+                        new UsesMethod<>("org.slf4j.Logger trace(..)"),
+                        new UsesMethod<>("org.slf4j.Logger debug(..)"),
+                        new UsesMethod<>("org.slf4j.Logger info(..)"),
+                        new UsesMethod<>("org.slf4j.Logger warn(..)"),
+                        new UsesMethod<>("org.slf4j.Logger error(..)")
+                ),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation mi, ExecutionContext ctx) {
+                        final int firstFormatArgIndex = mi.getArguments().get(0).getType().toString() == "org.slf4j.Marker" ? 2 : 1;
```

Comment:
Prefer using the methods available on `TypeUtils` to compare types, as those handle a couple more edge cases.

Also it's not safe to use `==` to compare Strings; does your IDE not give the same warning? Not saying to use `equals` here instead, but wondering how that slipped by. Are you on IntelliJ Ultimate with warnings enabled? We strive for zero.

Reply:
> Also it's not safe to use == to compare Strings; does your IDE not give the same warning? Not saying to use equals here instead, but wondering how that slipped by. Are you on IntelliJ Ultimate with warnings enabled? We strive for zero.

Of course, yes IDEA does show a warning, I'm quite aware of why this can't and shouldn't be done, it slipped off. I blame it on the past two years of using python where every comparison is using `== ` ;)